service.py

```diff
@@ -49,7 +50,7 @@ def do_GET(self):
             'simulate': True
         }
         if url.path == '/process':
-            ydl_opts['format'] = '(best[height = 1080][fps <= 30]/best[height <=? 720])[format_id!=source][vcodec!*=av01][vcodec!*=vp9]'
+            ydl_opts['format'] = f'{"bestaudio" if version and version[0] == "3" else "(best[height = 1080][fps <= 30]/best[height <=? 720])"}[format_id!=source][vcodec!*=av01][vcodec!*=vp9]'
```

Comment:
I think it's worth a comment explaining what's going on here. 

Prior to version 3, this filter statement causes yt-dlp to select the single best track for us. And we just use the url for that track. So the single best track is a combined audio/video track with some criteria for resolution, fps and vcodec. 

In version 3, because we want to return the urls for multiple tracks... we end up having to do our own picking, which happens in "processFormats" function. However, for audio... we always just return the url for the single best audio track, hence we just ask ytdlp for 'bestaudio'.

Reply:
Yeah I have no idea what is going on here